### PR TITLE
feat: add postgres response for trasaction related statements

### LINKED ISSUE #4553: Bypass unsupported statements for postgresql
### What type of enhancement is this?

API improvement

### What does the enhancement do?

While connecting with various tools like metabase, superset as postgres data source, we need to bypass some of the unsupported statements and return fixed response.

This is already done in mysql module as `federated.sql` and could be implemented similarly for postgresql.

### `SHOW default_transaction_isolation;`

```
 default_transaction_isolation 
-------------------------------
 read committed
```

### `SHOW client_encoding;`

```
 client_encoding 
-----------------
 UTF8
```

### `BEGIN`

```
BEGIN
```

### `ROLLBACK`

```
ROLLBACK
```

### `COMMIT`

```
COMMIT
```

### `SET TRANSACTION ISOLATION LEVEL READ COMMITTED`

```
SET
```

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

Fixes #4553 

## What's changed and what's your intention?

This patch partially improved our compatibility with Superset and Metabase postgres source. Note that we still need hidden table support and more for full compatibility 

- Respond transaction related statements: BEGIN/ROLLBACK/COMMIT/SET TRANSATION .../SHOW TRANSACTION ...
- Add some fixed response for `SHOW ...`
- Add built in function `current_schema()`
- Add built in function `pg_catalog.version()`
- Updated version in postgres startup

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/servers/src/postgres/fixtures.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        let resp = get_data("show client_encoding;", query_context.clone());
+        assert_eq!(1, resp.row_schema().len());
+        let resp = get_data("show standard_conforming_strings;", query_context.clone());
+        assert_eq!(1, resp.row_schema().len());
+        let resp = get_data("show default_transaction_isolation", query_context.clone());
+        assert_eq!(1, resp.row_schema().len());
+
+        assert!(process("SELECT 1", query_context.clone()).is_none());
+        assert!(process("SHOW TABLES ", query_context.clone()).is_none());
+        assert!(process("SET TIME_ZONE=utc ", query_context.clone()).is_none());
```
[the remark this answers is not in the document]
Comment (sunng87) [reply src/servers/src/postgres/fixtures.rs:165]: Postgres uses `SET TIME ZONE` for that. We can address it in future patches.

**Review by shuiyisong — APPROVED** [review]
LGTM

File: src/servers/src/postgres/fixtures.rs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+static SHOW_PATTERN: Lazy<Regex> = Lazy::new(|| Regex::new("(?i)^SHOW (.*?);?$").unwrap());
+static SET_TRANSACTION_PATTERN: Lazy<Regex> =
+    Lazy::new(|| Regex::new("(?i)^SET TRANSACTION (.*?);?$").unwrap());
+static TRANSACTION_PATTERN: Lazy<Regex> =
+    Lazy::new(|| Regex::new("(?i)^(BEGIN|ROLLBACK|COMMIT);?").unwrap());
+
+/// Process unsupported SQL and return fixed result as a compatibility solution
+pub(crate) fn process<'a>(
+    query: &str,
+    _query_ctx: QueryContextRef,
```
Comment (MichaelScofield) [inline src/servers/src/postgres/fixtures.rs:60]: The `_query_ctx` is not used for now, better remove it to save a clone.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**